### PR TITLE
feat(camera): add public API methods for camera configuration

### PR DESCRIPTION
**Pull request by RaHehl**
### Description of change

- Add update_camera_public() in ProtectApiClient supporting LED, HDR, video mode, mic volume, OSD settings, smart detect object and audio types, and LCD message via PATCH /v1/cameras/{id}
- Add Camera.set_*_public() convenience methods for all of the above
- Add PublicHdrMode enum (auto/on/off) and OsdOverlayLocation enum
- Fix _set_channel_parents model validator to use getattr() so it is safe for partial cameras constructed from the 16-field public API response (validate_assignment=True would otherwise crash)
- Derive HDR local state from the mode parameter rather than the PATCH response (hdrType != hdrMode in the public API schema)
- Use getattr(..., True) for has_smart_detect check so public-API cameras (where featureFlags lacks hasSmartDetect) default to allowed rather than raising AttributeError
- Accept both StrEnum instances and plain strings in audioTypes / objectTypes by using list() instead of [t.value for t in ...]
- Add unit tests covering all new methods and edge cases
### Pull-Request Checklist

- [ ] Code is up-to-date with the `main` branch
- [ ] This pull request follows the [contributing guidelines](https://github.com/uilibs/uiprotect/blob/main/CONTRIBUTING.md).
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/), such as "fix(api): prevent racing of requests".

> - If pre-commit.ci is failing, try `pre-commit run -a` for further information.
> - If CI / test is failing, try `poetry run pytest` for further information.

## Summary by CodeRabbit

* **New Features**
  * Public camera PATCH for many settings: name, HDR/video mode, LED controls, mic volume (validated 1–100), OSD toggles including overlay position, LCD message, and smart-detection object/audio type lists. Empty updates are rejected.
  * Added HDR mode and OSD overlay position enums and exports.

* **Tests**
  * New async test suite covering public camera endpoints, update payload shapes, validation, and device-level mutators.

[Review Change Stack](https://app.coderabbit.ai/change-stack/uilibs/uiprotect/pull/816?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by RaHehl** [comment]
@coderabbitai review

**Comment by bdraco** [comment]
@bluetoothbot review

**Comment by RaHehl** [comment]
The local state update is correct. The codebase uses two fields: hdr_mode (bool) and isp_settings.hdr_mode (enum). AUTO maps to hdr_mode=True + isp_settings.hdr_mode=NORMAL, which is exactly what hdr_mode_display expects to return "auto" — the same encoding the private API uses. The tri-state is fully preserved.

File: src/uiprotect/data/devices.py [hunk 1]
```diff
@@ -2832,6 +2850,276 @@ async def ptz_patrol_stop_public(self) -> None:
...
+        """Set HDR mode via public API."""
+        if not self.feature_flags.has_hdr:
+            raise BadRequest("Camera does not have HDR")
+        await self._api.update_camera_public(self.id, hdr_type=mode)
+        # The public API response uses 'hdrType', not 'hdrMode'/'ispSettings',
+        # so we derive the local state update directly from the mode we sent.
+        hdr_on = mode != PublicHdrMode.OFF
+        if hasattr(self, "hdr_mode"):
+            self.hdr_mode = hdr_on
+        isp = getattr(self, "isp_settings", None)
```
Comment (bdraco) [inline src/uiprotect/data/devices.py:2888]: is the hasattr/getattr needed here?

File: src/uiprotect/api.py [hunk 2]
```diff
@@ -3013,6 +3031,83 @@ async def get_camera_public(self, camera_id: str) -> Camera:
...
+            body["osdSettings"] = osd
+        if not body:
+            raise BadRequest("At least one parameter must be provided")
+        result = await self.api_request_obj(
+            url=f"/v1/cameras/{camera_id}",
+            method="patch",
+            json=body,
+            public_api=True,
+        )
+        return Camera.from_unifi_dict(**result, api=self)
```
Comment (bdraco) [inline src/uiprotect/api.py:3109]: its kind messy that we don't have a `PublicCamera`.. but thats more about the current state of the api with there being two APIs.. probably something we have to accept for now

**Review by bdraco — APPROVED** [review]
